src/model/selection.js

```diff
@@ -661,3 +705,30 @@ export default class Selection {
 }
 
 mix( Selection, EmitterMixin );
+
+// Checks whether the given element extends $block in the schema and has a parent (is not a root).
+// Marks it as already visited.
+function checkIsBlockContainer( element, visited ) {
```

Comment:
Hm... To my taste, this is too verbose, but I understand your decision and can live with it :D.